web-api/src/personalization/routes.rs

```diff
@@ -151,6 +167,154 @@ pub(crate) async fn update_interactions(
     Ok(())
 }
 
+#[derive(Deserialize)]
+struct StatelessPersonalizationRequest {
+    #[serde(default)]
+    count: Option<usize>,
+    #[serde(default)]
+    published_after: Option<DateTime<Utc>>,
+    history: Vec<UncheckedHistoryEntry>,
+}
+
+impl StatelessPersonalizationRequest {
+    fn resolved_history(
+        self,
+        config: &PersonalizationConfig,
+        warnings: &mut Vec<Warning>,
+    ) -> Result<Vec<HistoryEntry>, Error> {
+        let max_history_len = config.max_stateless_history_size;
+        if self.history.len() > max_history_len {
+            warnings.push(format!("history truncated, max length is {max_history_len}").into());
+        }
```

Comment:
❓ since we require in the spec that `history` is a non-empty array, don't we also have to check here that `!self.history.is_empty()`?

Reply:
yes, right there was one open question I had wrt. that:

- should it return a normal bad request or a not enough interactions error?

I guess a normal bad request?

Reply:
i'd say so too, if it doesn't conform to the spec, it's a bad request.